File: a2/src/variable.cc
#include "variable.h"

#include <string>
#include <vector>
#include <initializer_list>

Variable::Variable(std::string name, std::initializer_list<std::string> vals) : name_(name) {
  for (auto val : vals)
    vals_.push_back(val);
}

Variable::~Variable(void) {}

File: a7/src/lua/variable.lua
local Variable = {}

-- Number of variables created. A variable should only be created once.
local __g_var_c = 0

function Variable.create(n_vals)
  local v = {
    id = __g_var_c,  -- ID
    vals = {},       -- Possible values
    n = n_vals       -- Number of possible values
  }

  __g_var_c = __g_var_c + 1

  for i=1,n_vals,1 do
    v[i] = i-1
  end

  return v
end
